Gems/Atom/RPI/Code/Source/RPI.Builders/Model/ModelAssetBuilderComponent.cpp

```diff
@@ -1117,9 +1128,32 @@ namespace AZ
                 finalMeshList.reserve(mergedMeshCount + unmergeableMeshCount);
 
                 // Add the merged meshes
-                for (const auto& it : meshesByMatUid)
+                for (auto& it : meshesByMatUid)
                 {
-                    const ProductMeshContentList& meshList = it.second;
+                    ProductMeshContentList& meshList = it.second;
+
+                    for (auto meshIter = meshList.begin(); meshIter < meshList.end() - 1;)
+                    {
+                        // Any mesh that doesn't match the others will not be merged and will just be added directly to the final mesh list
+                        // This could result in multiple meshes that do not match the one before them, but still match each other, not
+                        // getting merged. But it's not worth over complicating things by trying to sort and split the meshList when the
+                        // common case is that everything assigned to the same material uid likely matches already anyways
+                        if (!VertexStreamLayoutMatches(*meshIter, *(meshIter + 1)))
+                        {
+                            if (MismatchedVertexLayoutsAreErrors())
```

Comment:
I think you can save the return of MismatchedVertexLayoutsAreErrors() in the beginning to avoid evaluate it many times in the for look.

Reply:
Yes, that makes sense. Will do.